polkadot/node/malus/src/variants/spam_statement_requests.rs

```diff
@@ -0,0 +1,160 @@
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A malicious node variant that attempts spam statement requests.
+//!
+//! This malus variant behaves honestly in everything except when propagating statement distribution
+//! requests through the network bridge subsystem. Instead of sending a single request when it needs
+//! something it attempts to spam the peer with multiple requests.
+//!
+//! Attention: For usage with `zombienet` only!
+
+#![allow(missing_docs)]
+
+use polkadot_cli::{
+	service::{
+		AuthorityDiscoveryApi, AuxStore, BabeApi, Block, Error, ExtendedOverseerGenArgs,
+		HeaderBackend, Overseer, OverseerConnector, OverseerGen, OverseerGenArgs, OverseerHandle,
+		ParachainHost, ProvideRuntimeApi,
+	},
+	validator_overseer_builder, Cli,
+};
+use polkadot_node_network_protocol::request_response::{outgoing::Requests, OutgoingRequest};
+use polkadot_node_subsystem::{messages::NetworkBridgeTxMessage, SpawnGlue};
+use polkadot_node_subsystem_types::DefaultSubsystemClient;
+use sp_core::traits::SpawnNamed;
+
+// Filter wrapping related types.
+use crate::{interceptor::*, shared::MALUS};
+
+use std::sync::Arc;
+
+/// Wraps around network bridge and replaces it.
+#[derive(Clone)]
+struct RequestSpammer<Spawner> {
+	spawner: Spawner, //stores the actual network bridge subsystem spawner
+	spam_factor: u32, // How many statement distribution requests to send.
+}
+
+impl<Sender, Spawner> MessageInterceptor<Sender> for RequestSpammer<Spawner>
+where
+	Sender: overseer::NetworkBridgeTxSenderTrait + Clone + Send + 'static,
+	Spawner: overseer::gen::Spawner + Clone + 'static,
+{
+	type Message = NetworkBridgeTxMessage;
+
+	/// Intercept NetworkBridgeTxMessage::SendRequests with Requests::AttestedCandidateV2 inside and
+	/// duplicate that request
+	fn intercept_incoming(
+		&self,
+		_subsystem_sender: &mut Sender,
+		msg: FromOrchestra<Self::Message>,
+	) -> Option<FromOrchestra<Self::Message>> {
+		match msg {
+			FromOrchestra::Communication {
+				msg: NetworkBridgeTxMessage::SendRequests(mut requests, if_disconnected),
+			} => {
+				// AttestedCandidateV2 requests arrive 1 by 1
+				if requests.len() == 1 {
+					// Check if the request is of the type AttestedCandidateV2
+					if let Requests::AttestedCandidateV2(req) = &requests[0] {
+						// Temporarily store peer and payload for duplication
+						let peer_to_duplicate = req.peer.clone();
+						let payload_to_duplicate = req.payload.clone();
+
+						// Duplicate the request spam_factor times and append to the list
+						for _ in 0..self.spam_factor - 1 {
+							let (new_outgoing_request, _) = OutgoingRequest::new(
+								peer_to_duplicate.clone(),
+								payload_to_duplicate.clone(),
+							);
+							let new_request = Requests::AttestedCandidateV2(new_outgoing_request);
+							requests.push(new_request);
+						}
```

Comment:
It should work without needing `peer_to_duplicate` and `payload_to_duplicate` clones:

```suggestion
  requests.extend(0..self.spam_factor - 1).into_iter().map(|_| 
                             Requests::AttestedCandidateV2(OutgoingRequest::new(
								peer_to_duplicate.clone(),
								payload_to_duplicate.clone(),
							)))
```

Reply:
Borrowed the extend syntax but still need the clones to satisfy the borrow checker.